project/src/main/java/com/google/sps/servlets/SearchDataServlet.java

```diff
@@ -0,0 +1,67 @@
+package com.google.sps.servlets;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ListMultimap;
+import com.google.sps.utilities.CommonUtils;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles search queries and addition of event keywords to index. */
+@WebServlet("/search-data")
+public class SearchDataServlet extends HttpServlet {
+  private ListMultimap<String, String> keywordToEventIds = ArrayListMultimap.create();
+  private static final String KEYWORD = "keyword";
+  private static final String EVENT_ID = "event-id";
+  private static final String DESCRIPTION = "description";
+  
+  /**
+   * Return search results for the keyword specified in the request.
+   *
+   * @param request servlet request
+   * @param response servlet response
+   * @throws IOException if Input/Output error occurs
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String keyword = request.getParameter(KEYWORD);
+    if (keyword == null) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, String.format("No keyword specified."));
+      return;
+    }
+
+    response.setContentType("application/json;");
+    response.getWriter().println(CommonUtils.convertToJson(keywordToEventIds.get(keyword)));
+  }
+
+  /**
+   * Add keywords in the description specified to the index with mapping to the event id.
+   *
+   * @param request servlet request
+   * @param response servlet response
+   * @throws IOException if Input/Output error occurs
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Is this called whenever the user presses 'search' or whenever an event is created? In the first case, how do you look at all events (there seems to be no db query)? in the second case, keywordToEventIds hold all of the event details of every event?

Reply:
First, I will get the algorithm working, then I will add persistent storage.

Reply:
Ok I see!